core-tests/shared/src/test/scala/zio/ZIOSpec.scala

```diff
@@ -1268,3 +1388,48 @@ object ZIOSpecHelper {
 
   def AsyncUnit[E] = IO.effectAsync[E, Unit](_(IO.unit))
 }
+
+object UncurriedBracketCompilesRegardlessOrderOfEAndRTypes {
```

Comment:
Can this be moved outside of spec file?

Reply:
Totally seprate file? BracketTypeInferenceSpec.scala?

Reply:
Yeah. Not that it matters much, but it's a bit awkward check to mix with regular specs.